src/ResourceManagement/Monitor/DiagnosticSettingImpl.cs

```diff
@@ -24,6 +24,7 @@ internal partial class DiagnosticSettingImpl :
         IDefinition,
         IUpdate
     {
+        internal static string SubscriptionDiagnosticSettingsUri = "/providers/AzureResourceManager/diagnosticSettings/";
```

Comment:
[Subscription Diagnostic Settings](https://docs.microsoft.com/en-us/rest/api/monitor/subscriptiondiagnosticsettings/createorupdate) is different from [Diagnostic Settings](https://docs.microsoft.com/en-us/rest/api/monitor/diagnosticsettings/createorupdate#diagnosticsettingsresource). The `Subscription Diagnostic Settings` is not supported as per [package-2019-06](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/monitor/resource-manager#tag-package-2019-06), which is the current version we use.

Reply:
I see, so this is not the right way to solve the issue?